### DISCUSSION TITLE
Replacing the current src with the next one from the src array in case of a download error

### DISCUSSION THREAD

**Discussion by maknakis** [vidstack/player#1414] [Q&A] [answered]
Hello everyone! 

I have a question: is it possible to set up the process of changing the source from the sources array using the vidstack api if the current source returns an error when downloading?

A brief process of how videos are uploaded to the my application: The user uploads the original video to s3 storage and saves a link to it in the DB. After the download is complete, the video is sent to the microservice for transcoding into hls format in 4 different resolutions. After transcoding, a link to the hls master playlist is written to the DB.

In the context of my application, the following situation may arise:
The links are signed and may expire. Therefore, when playing, the player will return an error that the video cannot be played. I would like to set up a process for switching sources to the next one in the list if an error occurs with playback.

I will be glad to any help or advice!

[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)][stackblitz-demo]

[stackblitz-demo]: https://stackblitz.com/edit/vidstack-examples-2pu9wc

**Comment by mihar-22** [comment, marked as answer]
Hey, @maknakis!

First, don't pass the array of sources to the player, rather just an index that points to a single source you want to play. Inside the `onError` callback, increment the index to go to the next source.

```tsx
const [srcIndex, setSrcIndex] = useState(0);

function onError() {
  // try next source (need to consider what to do when no sources left).
  setSrcIndex(n => n + 1);
}

<MediaPlayer src={sources[index]} onError={onError}>
```

**Reply by maknakis** [reply]
Thanks for your option, @mihar-22! Simple and clear.

Am I right in understanding that you can't control the current source via api?

**Reply by mihar-22** [reply]
Not at the moment other than changing the `src` player property.
